src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -175,21 +175,29 @@ public void visitToken(DetailAST ast) {
 
         if (rcurly != null) {
             final String violation;
-            if (shouldStartLine) {
-                final String targetSourceLine = getLines()[rcurly.getLineNo() - 1];
-                violation = validate(details, option, true, targetSourceLine);
-            }
-            else {
-                violation = validate(details, option, false, "");
-            }
-
+            violation = validate(details, option,
+                    shouldStartLine, getTargetSourceLine(rcurly.getLineNo() - 1));
             if (!violation.isEmpty()) {
                 log(rcurly, violation, "}", rcurly.getColumnNo() + 1);
             }
         }
     }
 
     /**
+     * Fetches line based on passed line no.
+     * @param lineNo for fetching lines.
+     * @return lines associated with line no.
+     */
+    private String getTargetSourceLine(int lineNo) {
```

Comment:
@sagar-shah94 ,

ok, .....

please make `validate` as non static. 
`validate` should have only one argument "details", all other should be taken from class fields and methods ... .